decidim-elections/app/commands/decidim/elections/admin/mark_answer_as_selected.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Elections
+    module Admin
+      # This command is executed when an admin marks an answer
+      # as selected.
+      class MarkAnswerAsSelected < Rectify::Command
```

Comment:
When I first read the name for this command, I was expecting to find a `MarkAnswerAsUnselected`. Then I reviewed the code and I understood that we can use this one for that too. I browsed the project code to find similar scenarios like this, but I didn't find anything. What I found were many situations where there are two commands (vote/unvote, publish/unpublish, endorse/unendorse), but they are different because they have a more complex logic.

Also, I think the way you are toggling between selected and not selected values is very clever, but it's a pity that is not [idempotent](https://en.wikipedia.org/wiki/Idempotence). This is a very desired property for commands, and it means that it should be equivalent to apply them once, twice or many times. In this case, each time you execute the command the result will be different (selected, not selected, selected, ...). With that, you can think what would happen if two admin users were trying to apply the same change one after another.

So, to sum up, I would use a name like `UpdateAnswerSelection` and receive the new value for the selected field as a parameter to ensure their idemponce. What do you think?

Reply:
Thanks @leio10. For the comment and the whole explanation! 👍 I agree to 100% and will update the command!